src/Microsoft.AspNetCore.Mvc.Abstractions/Formatters/InputFormatterContext.cs

```diff
@@ -67,10 +97,20 @@ public class InputFormatterContext
             ModelState = modelState;
             Metadata = metadata;
             ReaderFactory = readerFactory;
+            AllowEmptyInput = allowEmptyInput;
             ModelType = metadata.ModelType;
         }
 
         /// <summary>
+        /// Gets a flag to indicate whether the input formatter should allow no value to be provided.
+        /// If <see langword="false"/>, the input formatter should handle empty input by returning
+        /// <see cref="InputFormatterResult.NoValueAsync()"/>. If <see langword="false"/>, the input
+        /// formatter should handle empty input by returning the default value for the type
+        /// <see cref="ModelType"/>.
```

Comment:
> Need a different name in this context

OK, renamed to `TreatEmptyInputAsDefaultValue`.

> One of these sentences should describe the true case

Good catch - fixed!

> Don't think we use langword elsewhere

We use it [pretty widely](https://github.com/search?q=org%3Aaspnet+langword&type=Code) so I'll leave it as-is. I guess the benefit over `<c>false</c>` is that IDEs and autogenerated docs know it's something they can optionally translate to the language that the user is working in (e.g., it could be presented as `False` in a VB project (if we supported VB)).